src/components/IdentityView/IdentityView.tsx

```diff
@@ -79,45 +81,49 @@ class IdentityView extends React.Component<Props, State> {
         <div className="attributes">
           <div>
             <label>Alias</label>
-            <div>{myIdentity.metaData.name}</div>
+            <div>{metaData.name}</div>
           </div>
           <div>
             <label>Phrase</label>
-            <div>{myIdentity.phrase}</div>
+            <div>{phrase}</div>
           </div>
           <div>
             <label>KILT Address</label>
-            <div>{myIdentity.identity.address}</div>
+            <div>{identity.address}</div>
           </div>
           <div>
             <label>Seed (as hex)</label>
-            <div>{myIdentity.identity.seedAsHex}</div>
+            <div>{identity.seedAsHex}</div>
           </div>
           <div>
             <label>Public Key</label>
-            <div>{myIdentity.identity.signPublicKeyAsHex}</div>
+            <div>{identity.signPublicKeyAsHex}</div>
           </div>
           <div>
             <label>Encryption Public Key</label>
-            <div>{myIdentity.identity.boxPublicKeyAsHex}</div>
+            <div>{identity.boxPublicKeyAsHex}</div>
           </div>
           <div>
             <label>DID</label>
             <div>
-              {myIdentity.did ? (
-                <span className="did">{myIdentity.did}</span>
+              {did ? (
+                <span className="did">
+                  <DidView did={did.document}>{did.address}</DidView>
+                </span>
               ) : (
-                ''
+                <>
+                  <div>Identity doesn't own a DID.</div>
```

Comment:
Suggestion: 
No DID is attached to this identity

Reply:
Updated!